trappy/base.py

```diff
@@ -200,13 +202,20 @@ def generate_parsed_data(self):
             # When running out of memory, Pandas has been observed to segfault
             # rather than throwing a proper Python error.
             # Look at how much memory our process is using and warn if we seem
-            # to be getting close to the system's limit.
+            # to be getting close to the system's limit, check it only once
+            # in the beginning and then every CHECK_MEM_COUNT events
+            check_memory_count -= 1
+            if check_memory_count != 0:
+                yield data_dict
+                continue
```

Comment:
I think it would be clearer if you inverted this `if` and partially combined it with the one below, like:
```
check_memory_count -= 1
if check_memory_usage and check_memory_count == 0:
    kb_used = (getrusage(RUSAGE_SELF).ru_maxrss - starting_maxrss)
    if kb_free and kb_used > kb_free * 0.9:
        warnings.warn("TRAPpy: Appear to be low on memory. "
                               "If errors arise, try providing more RAM")
        check_memory_usage = False
    check_memory_count = CHECK_MEM_COUNT
```